### PR TITLE
feat: add Imperaitive Handle to public API

### PR DESCRIPTION
**Pull request by Caele**
## Motivation

Exposes the Imperative handle API in the form of
- useImperativeHandle component hook
- getimperativeHandle viz API function

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [ ] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [x] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/supernova/src/hooks.js [hunk 1]
```diff
@@ -983,15 +983,11 @@ export function useOptions() {
...
- * Exposes an API to the external environment.
- *
  * This is an empty object by default, but enables you to provide a custom API of your visualization to
  * make it possible to control after it has been rendered.
  *
  * You can only use this hook once, calling it more than once is considered an error.
  * @entry
- * @private
  * @template T
  * @param {function():T} factory
```
Comment (T-Wizard) [inline apis/supernova/src/hooks.js:992]: maybe simplify to
```suggestion
 * @param {function():object} factory
```
The generic type T don't give any value

**Comment by veinfors** [comment]
Looks good to me.
A good idea is to test this change with some existing charts where the `useImperativeHandle` hook is used already, like the sn-bar-chart, sn-pie-chart etc. to make sure we don't break anything for existing extensions.

File: apis/supernova/src/hooks.js (hunk 1 above)
Comment (Caele) [reply apis/supernova/src/hooks.js:992]: hmm, you could have the handle function return whatever you want, shouldn't it be the generic type then? Not sure how a TS setup handles it?

**Comment by Caele** [comment]
> Looks good to me. A good idea is to test this change with some existing charts where the `useImperativeHandle` hook is used already, like the sn-bar-chart, sn-pie-chart etc. to make sure we don't break anything for existing extensions.

There shouldn't be any functional changes, only typings are added and the API exposed to mashups.

**Review by veinfors — APPROVED** [review]

**Review by T-Wizard — APPROVED** [review]

File: apis/supernova/src/hooks.js (hunk 1 above)
Comment (T-Wizard) [reply apis/supernova/src/hooks.js:992]: as this is the only place where the type is referenced I don't think it gives any value.
(but I also don't there is any real problem with keeping it as it is)

**Pull request merged**